backend/cdApp.py

```diff
@@ -230,45 +211,44 @@ def post(self, course_id, operation):
             if cursor.rowcount == 0:
                 raise InvalidUsage('Entry not found in database. Please use `setpiazza` instead')
             return jsonify(message='Successfully updated piazza ID for course', course_id=course_id)
+
         else:
             raise InvalidUsage('Unknown operation type')
 
     def get(self, course_id, operation):
-        if operation != 'getpiazza':
-            raise InvalidUsage('Unknown operation type')
+        if operation == 'getpiazza':
+            cursor = g.db.execute('''SELECT piazza_cid
+                                     FROM courses
+                                     WHERE course_id=(?)''',
+                                  [int(course_id)])
+            piazza_id_str = cursor.fetchone()
+            if piazza_id_str is None:
+                raise InvalidUsage('Given course does not have a Piazza ID')
+            else:
+                return jsonify(message='Returning piazza ID for course', course_id=course_id, piazza_cid=piazza_id_str)
 
-        cursor = g.db.execute('''SELECT piazza_cid
-                                 FROM courses
-                                 WHERE course_id=(?)''',
-                              [int(course_id)])
-        piazza_id_str = cursor.fetchone()
-        if piazza_id_str is None:
-            raise InvalidUsage('Given course does not have a Piazza ID')
+        elif operation == 'getpiazzaposts':
+            cursor = g.db.execute('''SELECT piazza_cid
+                                     FROM courses
+                                     WHERE course_id=(?)''',
+                                  [int(course_id)])
+            piazza_id_str = cursor.fetchone()['piazza_cid']
+            if piazza_id_str is None:
+                raise InvalidUsage('Given course does not have a Piazza ID')
+            else: #TODO: handle errors
+                print piazza_id_str
+                p = Piazza()
+                p.user_login(email="sakekasi@ucla.edu", password="password")
```

Comment:
Anyway to get this out of the code? Even if we use dummy passwords, it's still probably a better idea to read it in from disk.

```Python
with open(CREDENTIALS_FILE, 'r') as fname:
    lines = fname.read().split()
    my_email = lines[0]
    my_pass = lines[1]
    p.user_login(email=my_email, password=my_pass)
```

Should be close to what you want, where `CREDENTIALS_FILE` is saved on the system (or you can save it in a DB if that makes more sense).